logging/logging.go

```diff
@@ -298,13 +298,75 @@ func detectGAEResource() *mrpb.MonitoredResource {
 	}
 }
 
+func isCloudRun() bool {
+	_, config := os.LookupEnv("K_CONFIGURATION")
+	return config
+}
+
+func detectCloudRunResource() *mrpb.MonitoredResource {
+	projectID, err := metadata.ProjectID()
+	if err != nil {
+		return nil
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		return nil
+	}
+	return &mrpb.MonitoredResource{
+		Type: "cloud_run_revision",
+		Labels: map[string]string{
+			"project_id":         projectID,
+			"location":           regionFromZone(zone),
+			"service_name":       os.Getenv("K_SERVICE"),
+			"revision_name":      os.Getenv("K_REVISION"),
+			"configuration_name": os.Getenv("K_CONFIGURATION"),
+		},
+	}
+}
+
+func isCloudFunction() bool {
+	_, name := os.LookupEnv("FUNCTION_NAME")
```

Comment:
These variables seem pretty generic. Users on other platforms might have their own `FUNCTION_NAME` set by coincidence, and have their code interpreted as running on Cloud Functions.

In Python, I do this:
```
_CLOUD_RUN_SERVICE_ID = "K_SERVICE"
_FUNCTION_TARGET = "FUNCTION_TARGET"
_FUNCTION_SIGNATURE = "FUNCTION_SIGNATURE_TYPE"
_FUNCTION_NAME = "FUNCTION_NAME"
_FUNCTION_REGION = "FUNCTION_REGION"
_FUNCTION_ENTRY = "ENTRY_POINT"
_FUNCTION_ENV_VARS = [_FUNCTION_TARGET, _FUNCTION_SIGNATURE, _CLOUD_RUN_SERVICE_ID]
_LEGACY_FUNCTION_ENV_VARS = [_FUNCTION_NAME, _FUNCTION_REGION, _FUNCTION_ENTRY]

if all([env in os.environ for env in _LEGACY_FUNCTION_ENV_VARS]) or all(
        [env in os.environ for env in _FUNCTION_ENV_VARS]):
   return _create_functions_resource()
```

Reply:
Great point. I'll follow suite

Reply:
suit*